packages/framework/synthesize/src/dependencyContainer.ts

```diff
@@ -95,69 +103,94 @@ export class DependencyContainer implements IFluidDependencySynthesizer {
             return provider;
         }
 
-        if (this.parent) {
-            return this.parent.getProvider(type);
+        for(const parent of this.parents) {
+            const p = parent.getProvider(type);
+            if(p !== undefined) {
+                return p;
+            }
         }
 
         return undefined;
     }
 
     private generateRequired<T extends IFluidObject>(
+        base: AsyncRequiredFluidObjectProvider<FluidObjectKey<T>>,
```

Comment:
Rather than modify a param, my preference here would be to collapse `generateRequired` and `generateOptional` into `synthesize` -- they aren't called anywhere else and aren't overly long.

Reply:
yeah. there is another set of changes coming. once i no longer expose providers directly i can just create the `PropertyDescriptions` upfront on register, and then all this code will be way simpler. I do those change with what you suggest in the next release